### ISSUE TITLE
Settings not save in prefab

### ISSUE THREAD

**Issue by JabluszkoPL** [Microsoft/AppCenter-SDK-Unity#377]
### **Description**

If I am manualy spawning AppCenterBehavior object from prefab it dose not have setting value. Your code is setting it after each time when I am focusing on prafab - but unity is not see any changes it.

### **Repro Steps**

1. Create empty gameobject in scene add AppCenterBehavior script
2. Create prefab from this gameobject
3. Remove this prefab from scene.
4. Add reference to prefab in some class that will spawn it on game start
5. Remove setting
6. Create new setting by focusing on prefab
6. Restart editor (in this point prefab will lose new reference to setting)
7. run editor and game
8. You will get error that setting are null

### **How to fix this**

Add 2 lines when you are setting settings in AppCenterBehaviorEditor.cs:
Original code:
`if (behaviour.Settings == null)
{
    behaviour.Settings = AppCenterSettingsContext.SettingsInstance;
}`

to:
`if (behaviour.Settings == null)
{
    behaviour.Settings = AppCenterSettingsContext.SettingsInstance;
    EditorUtility.SetDirty(behaviour);
    EditorSceneManager.MarkSceneDirty(behaviour.gameObject.scene);
}`
then unity will see changes in files.

### **Details**

1. Which SDK version are you using?
    - 2.4.0
2. Which platform did you experience the issue on?
    - Window, Mac, Android, iOS
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - editor and all builds

**Comment by annakocheshkova** [comment]
Hi @JabluszkoPL to be honest, I don't completely understand your use case.
As I assume, you're trying to instantiate Setting object dynamically at the game start (may I ask, why?). This won't work as is, because we have the native start logic and you'll have to [delay the start](https://docs.microsoft.com/en-us/appcenter/sdk/other-apis/unity#delay-the-start-of-app-center) if you do this.

So if I follow your steps, then start the app, for example, on iOS, I do get errors that no services are started (because of what I said above), but I don't get the error that **Settings** are null.
That's the code I used:
```
using UnityEngine;
public class NewMonoBehaviour : MonoBehaviour
{
    public GameObject myPrefab;

    void Start()
    {
        Instantiate(myPrefab);
    }
}
```

And that's how it looks like:
![image](https://user-images.githubusercontent.com/34677812/69733037-c12f4800-113d-11ea-8a16-e6f6ea094ab4.png)

**Comment by JabluszkoPL** [comment]
Hu @annakocheshkova 

> As I assume, you're trying to instantiate Setting object dynamically 
No, I am instancing prefaba like you shown but thing is that if you will remove setting (not game objects but settings)and then focus this prefab it will create new settings file, and the problem is here  this new config will not attach to this prefab - after close and open Unity this new setting will not apply - you can check this by running game and seeing error that there is not settings config

**Comment by annakocheshkova** [comment]
@JabluszkoPL I did as you said in the repro steps, then I restarted the Editor, built the app for iOS, and run. I don't see the error anywhere in the logs (neither in the Build Logs, nor in the Console after starting the app).
Here's a screen recording with my steps: [one-minute video](https://i.imgur.com/KSiuipW.mp4), and the code I already shared above.
Is that what you do to get this error?

**Comment by JabluszkoPL** [comment]
If you would test it like in this video then You would get error:
> App Center Behavior should have only one instance!

I am sending You my test project with this issue: https://drive.google.com/file/d/1w-IvvnKuo8If9qvO5e64YddN-vI0Au6J/view?usp=sharing

1)  run the game in editor - you will get error with missing config
2) focus GameObject
3) notice that nothing changed
4) run game -> no error
5) save and reopen editor (you can also build game - after build it will act like editor reset)
6) run game -> notice same error

**Comment by annakocheshkova** [comment]
Thanks, let me try and get back to you.

**Comment by annakocheshkova** [comment]
@JabluszkoPL I am able to see the error now, thanks. And seems like I follow your use case: you don't want to have an App Center game object directly in the scene. 

I will file a ticket for the fix you proposed, but if you want to submit a PR - contributions are highly appreciated and welcome (and this will more likely be released soon)! 🙂

**Comment by JabluszkoPL** [comment]
@annakocheshkova  I mad pull request - https://github.com/microsoft/appcenter-sdk-unity/pull/382

**Comment by AnastasiaKubova** [comment]
The fix had been released in [2.6.0](https://github.com/microsoft/appcenter-sdk-unity/releases/tag/2.6.0).
